numpy/__init__.pyi

```diff
@@ -541,6 +552,11 @@ class ndarray(_ArrayOrScalarCommon, Iterable, Sized, Container):
 # See https://github.com/numpy/numpy-stubs/pull/80 for more details.
 
 _CharLike = Union[str, bytes]
+_BoolLike = Union[bool, bool_]
+_IntLike = Union[int, integer]
+_FloatLike = Union[int, integer, float, floating]
+_ComplexLike = Union[int, float, complex, number]
```

Comment:
I'm wary of using `number` in this last one. I'd be inclined to write this
```suggestion
_IntLike = Union[int, integer]
_FloatLike = Union[_IntLike, float, floating]
_ComplexLike = Union[_FloatLike, complex, complexfloating]
```
While all numbers may be coercible to complex today, third-party types like `quaternion` will not be.

Reply:
Fair point, better to be on the safe side here. Implemented in https://github.com/numpy/numpy/pull/17273/commits/0e4a26771e787cc85232965cdfea6592bcef1526.

As a side note: there is a limit to allowed levels of `typing.Union` nesting, which is why I chose not to reuse `_IntLike` and `FloatLike` in the definition of `_ComplexLike`, despite the minor increase in code duplication.

Reply:
I assume that limit is a mypy issue? That sucks. Mind adding a comment to that effect in the code?

Reply:
While I recall seeing this issue in the past I can't seem to reproduce it anymore; I suspect it has been fixed in one of the more recent mypy releases?

In any case, https://github.com/numpy/numpy/pull/17273/commits/4ab85b4e7da0b3e48d519cdc173a57234868e9ce introduced the change you initially suggested. 
If, at some point, any nesting related issues appear after all then we can always get rid of the nesting.